internal/provider/network_connection_models.go

```diff
@@ -0,0 +1,295 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/doublecloud/go-genproto/doublecloud/network/v1"
+	dcgennet "github.com/doublecloud/go-sdk/gen/network"
+	"github.com/hashicorp/terraform-plugin-framework-validators/objectvalidator"
+	dataschema "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	resourceschema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type NetworkConnectionModel struct {
+	ID          types.String `tfsdk:"id"`
+	NetworkID   types.String `tfsdk:"network_id"`
+	Description types.String `tfsdk:"description"`
+
+	AWS    *awsNetworkConnectionInfo    `tfsdk:"aws"`
+	Google *googleNetworkConnectionInfo `tfsdk:"google"`
+
+	status       string
```

Comment:
Please name all model fields of structure with capital letter.
@leskin-in said that it could introduce some errors in terraform.